windows/src/main/csharp/ch/cyberduck/ui/ViewModels/TransfersViewModel.cs

```diff
@@ -0,0 +1,535 @@
+        this.iconProvider = iconProvider;
+
+        var localTransfers = store.Transfers.Connect().ObserveOnDispatcher()
+            .Transform(m => new TransferViewModel(controller, m, locale)).DisposeMany()
+            .Bind(out transfers)
+            .AutoRefresh(m => m.IsSelected).Filter(m => m.IsSelected).Bind(out selectedTransfers)
+            .AsObservableCache();
+        localTransfers.Connect().WhenAnyPropertyChanged().Subscribe(OnSelectedTransferPropertyChanged);
+        localTransfers.CountChanged.Subscribe(OnSelectionCountChanged);
+
+        BandwidthViewModel unlimitedBandwidth = new(-1, true, locale.localize("Unlimited Bandwidth", "Preferences"));
+        Bandwidth = [
+            unlimitedBandwidth,
+            ..bandwidthProvider.Bandwidth.Select(m => new BandwidthViewModel(m, true)),
+        ];
+        bandwidthLookup = Bandwidth.ToDictionary(m => m.Bandwidth);
+        selectedBandwidth = unlimitedBandwidth;
+
+        Connections = [
+            .. connectionProvider.Connections.Select(m => new ConnectionViewModel(m, true)),
+        ];
+        connectionLookup = Connections.ToDictionary(m => m.Connections);
+        selectedConnectionLimit = FetchConnectionLimit();
+    }
+
+    partial void OnSelectedConnectionLimitChanged(ConnectionViewModel value)
+    {
+        preferences.setProperty("queue.connections.limit", value.Connections);
+        TransferQueueFactory.get().resize(value.Connections);
+    }
+
+    partial void OnSelectedTransferChanged(TransferViewModel newValue)
+    {
+        UpdateSelectedBandwidth();
+        UpdateSelectionPanel();
+    }
+
+    private ConnectionViewModel FetchConnectionLimit()
+    {
+        var rate = preferences.getInteger("queue.connections.limit");
+        if (!connectionLookup.TryGetValue(rate, out var selection))
+        {
+            connectionLookup[rate] = selection = new(rate, false, null);
+            Connections.Add(selection);
+        }
+
+        return selection;
+    }
+
+    [RelayCommand]
+    private void OnClean() => store.CleanCompleted();
+
+    [RelayCommand(CanExecute = nameof(ValidateOpenCommand))]
+    private void OnOpen()
+    {
+        var launcher = ApplicationLauncherFactory.get();
+        foreach (var file in SelectedTransfer.Roots)
+        {
+            if (launcher.open(file.TransferItem.Local))
+            {
+                return;
+            }
+        }
+    }
+
+    [RelayCommand(CanExecute = nameof(ValidateReloadCommand))]
+    private void OnReload()
+    {
+        foreach (var item in SelectedTransfers)
+        {
+            if (!item.Running)
+            {
+                TransferOptions options = new TransferOptions().resume(false).reload(true);
+                controller.StartTransfer(item.Transfer.Model, options);
+            }
+        }
+    }
+
+    [RelayCommand(CanExecute = nameof(ValidateRemoveCommand))]
+    private void OnRemove()
+    {
+        using var copy = SelectedTransfers.CopyToPool(true);
```

Comment:
While we are removing transfers, these are _immediately_ synced back to SelectedTransfers (due to them being removed), which would invalidate the enumerator, thus create temporary pooled copy.